documentation/staging/content/managing-domains/accessing-the-domain/domain-events.md

```diff
@@ -29,7 +29,8 @@ The domain events have been enhanced in 4.0. Here is a summary of the changes in
     * Have a better failure categorization (see [Operator-generated event types](#operator-generated-event-types) for more details).
     * Include the categorization information in the event message.
     * Provide more information in the event message to indicate what has gone wrong, what you need to do to resolve the problem, and if the operator will [retry]({{< relref "/managing-domains/domain-lifecycle/retry.md" >}}) the failed operation.
-* Added three event types, `Unavailable`, `Incomplete`, and `FailureResolved`, to record the transition of domain condition `Available/Completed/Failed` from `True` to `False`, and vice versa.
+* Added three event types, `Unavailable`, `Incomplete`, and `FailureResolved`, to record the transition of [domain condition]({{< relref "/managing-domains/accessing-the-domain/status-conditions.md" >}})
+  `Available/Completed/Failed` from `True` to `False`, and vice versa.
```

Comment:
The event types (at least the names) are defined here: https://github.com/oracle/weblogic-kubernetes-operator/blob/main/operator/src/main/java/oracle/kubernetes/operator/EventConstants.java. @doxiao or @russgold, do we have a design wiki that describes the events that would help Anthony fill out the documentation?

Reply:
> Only Available and Complete transition between True and False, where-as Failed is removed instead of being set to False. Furthermore, from what I can tell (and I might be wrong), there are actually a total of 10 event types - not three. 

In 4.0, we have 12 domain events, and 7 cluster events.

>How about:
> 
>     * Added six event types `Available`, `Unavailable`, `Complete`, `Incomplete`, `Failed`, and `FailureResolved` to record transitions of their corresponding three domain resource status [conditions].
> 
Completed and Failed are not new events, but renamed old events.

>     * Added four event types, `ClusterAvailable`, `ClusterUnavailable`, `ClusterComplete`, and `ClusterIncomplete` to record transitions of their corresponding two cluster resource status [conditions].

All 7 cluster events are new in 4.0. We need to add ClusterCreated, ClusterChanged, and ClusterDeleted as well.

Reply:
> The event types (at least the names) are defined here: https://github.com/oracle/weblogic-kubernetes-operator/blob/main/operator/src/main/java/oracle/kubernetes/operator/EventConstants.java. @doxiao or @russgold, do we have a design wiki that describes the events that would help Anthony fill out the documentation?

I am not aware of any design wiki for the 4.0 events.  We have a wiki page that tracks the initial event rework work in 4.0, and the contents in that wiki page has been reflected in the earlier version of the 4.0 event doc already, for example, rename of most of the events. A couple of events, such as domain Unavailable and Incomplete, have been added since then to record the transitions of domain conditions.